tests/credential_plugins_test.py

```diff
@@ -134,6 +135,75 @@ def test_hashivault_handle_auth_not_enough_args() -> None:
         hashivault.handle_auth()
 
 
+@pytest.mark.parametrize(
+    'kwargs', (
+        (
+            {
+                'access_key': 'my_access_key',
+                'secret_key': 'my_secret_key',
+                'role_arn': 'the_arn',
+                'identifier': 'access_token',
+            },
+        ),
+        (
+            {
+                'role_arn': 'the_arn',
+                'identifier': 'access_token',
+            },
+        ),
+    ),
+)
```

Comment:
This seems too nested:
```suggestion
    'kwargs',
    (
        {
            'access_key': 'my_access_key',
            'secret_key': 'my_secret_key',
            'role_arn': 'the_arn',
            'identifier': 'access_token',
        },
    ),
    (
        {
            'role_arn': 'the_arn',
            'identifier': 'access_token',
        },
    ),
)
```
Also, perhaps rename the argument to be more explicit about what it is?